pyproject.toml

```diff
@@ -43,7 +43,11 @@ select = ["E", "W", "F", "C", "N", "R", "D", "H"]
 # Ignore D107 Missing docstring in __init__
 ignore = ["W503", "E501", "D107"]
 # D100, D101, D102, D103: Ignore missing docstrings in tests
-per-file-ignores = ["tests/*:D100,D101,D102,D103,D104,D205,D212,D415,DC0020"]
+per-file-ignores = [
+    "src/ingress_definition.py:DCO060",
+    "src/controller.py:DCO020,DCO030,DCO031,DCO050,DCO055,DCO060",
```

Comment:
Could you check whether these are still needed?

Reply:
Okay, I have checked, and the `DCO020` `DCO030` and `DCO060` can be removed. But the rest has something to do with the abstract method don't have a `return` statement, and the problem with re-raising exception with a `raise` statement.